### ISSUE TITLE
[2017.7.0] python API can read custom pillar

### ISSUE THREAD

**Issue by blackmaz** [saltstack/salt#42591]
### Description of Issue/Question
I used a custom pillar to pass variables to sls. Custom pillar does not work when upgraded to 2017.7.0.

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

tmp/custompillar.sls
```
{%- set company = salt['pillar.get']('company','default') %}
{%- set system  = salt['pillar.get']('system','default') %}

test_pillar:
  cmd.run:
    - name: ls -l {{ company }}

test_pillar_2:
  cmd.run:
    - name: ls -l {{ system }}
```
### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)
It works well in 2016.11.6.
```
$ python
Python 2.7.12 (default, Nov 19 2016, 06:48:10)
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import salt.client
>>> localClient = salt.client.LocalClient()
>>> localClient.cmd('test1', 'state.show_sls', ['tmp.custompillar','pillar={"company": "p33", "system": "petclinic"}'])
{'test1': {'test_pillar_2': {'__env__': 'base', 'cmd': [{'name': 'ls -l petclinic'}, 'run', {'order': 10001}], '__sls__': 'tmp.custompillar'}, 'test_pillar': {'__env__': 'base', 'cmd': [{'name': 'ls -l p33'}, 'run', {'order': 10000}], '__sls__': 'tmp.custompillar'}}}
>>>
```
However, it will not work in 2017.7.0.
```
$ python
Python 2.7.12 (default, Nov 19 2016, 06:48:10)
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>>
>>> import salt.client
>>> localClient = salt.client.LocalClient()
>>> localClient.cmd('de2oall_test', 'state.show_sls', ['tmp.custompillar','pillar={"company": "p33", "system": "petclinic"}'])
{'de2oall_test': {'test_pillar_2': {'__env__': 'base', 'cmd': [{'name': 'ls -l default'}, 'run', {'order': 10001}], '__sls__': 'tmp.custompillar'}, 'test_pillar': {'__env__': 'base', 'cmd': [{'name': 'ls -l default'}, 'run', {'order': 10000}], '__sls__': 'tmp.custompillar'}}}
>>>
```
### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)

Version information for the non-working server.
```
$ salt --versions
Salt Version:
           Salt: 2017.7.0

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.4.3
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-83-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

The version information of the server that works well.
```
salt --versions
Salt Version:
           Salt: 2016.11.6

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.8.0-58-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by gtmanfred** [comment]
This is a known issue, we are working on getting a 2017.7.1 out to fix this.

Thanks,
Daniel

**Comment by gtmanfred** [comment]
I can't find where it is fixed, but on the 2017.7.1 branch, it is working correctly.

```
[root@salt salt]# tail -c +0 /srv/salt/test.sls this
==> /srv/salt/test.sls <==
{%- set company = salt['pillar.get']('company','default') %}
{%- set system  = salt['pillar.get']('system','default') %}

test_pillar:
  cmd.run:
    - name: ls -l {{ company }}

test_pillar_2:
  cmd.run:
    - name: ls -l {{ system }}

==> this <==
import salt.client
local = salt.client.LocalClient()
print local.cmd('*', 'state.show_sls', ['test','pillar={"company": "p33", "system": "petclinic"}'])
[root@salt salt]# python this
{'salt.c.gtmanfred-1263.internal': {'test_pillar_2': {'__env__': 'base', 'cmd': [{'name': 'ls -l petclinic'}, 'run', {'order': 10001}], '__sls__': 'test'}, 'test_pillar': {'__env__': 'base', 'cmd': [{'name': 'ls -l p33'}, 'run', {'order': 10000}], '__sls__': 'test'}}}
```

Thanks,
Daniel
